cdci_data_analysis/flask_app/app.py

```diff
@@ -298,13 +289,22 @@ def test_mock():
 
 @app.route('/resolve-job-url', methods=['GET'])
 def resolve_job_url():
-    logger.info('\033[32m===========================> resolve_job_url\033[0m')
-
-    query = InstrumentQueryBackEnd(app, instrument_name='mock', resolve_job_url=True)
-    location = query.resolve_job_url()
+    try:
+        logger.info('\033[32m===========================> resolve_job_url\033[0m')
 
-    return redirect(location, 302)
+        query = InstrumentQueryBackEnd(app, instrument_name='mock', resolve_job_url=True)
+        location = query.resolve_job_url()
 
+        return redirect(location, 302)
+    except APIerror as e:
+        message_dict = {
+            "error_message": e.message,
+            "status_code": e.status_code
+        }
+        location = '%s?%s' % (app.config['conf'].products_url, urlencode(message_dict))
+        return redirect(location, 302)
+    except Exception as e:
```

Comment:
when does this happen? in unexpected error? And then it does what, just redirects to bare site?

Reply:
Mainly is there to always redirect to the frontend page, but of course an addition more informative message is necessary

Reply:
Indeed! Would be quite unfortunate to click on the link and and up on the front page with no obvious error. It's bad to show incomprehensive errors, but it's even worse to pretend there is no error and do nothing! As I user I could be really annoyed.

Reply:
Now more error information are included in the redirect url